src/runtime/containerd-shim-v2/create.go

```diff
@@ -259,3 +272,104 @@ func doMount(mounts []*containerd_types.Mount, rootfs string) error {
 	}
 	return nil
 }
+
+func configureNonRootHypervisor(runtimeConfig *oci.RuntimeConfig) error {
+	userName, err := createVmmUser()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			removeVmmUser(userName)
+		}
+	}()
+
+	u, err := user.Lookup(userName)
+	if err != nil {
+		return err
+	}
+
+	uid, err := strconv.Atoi(u.Uid)
+	if err != nil {
+		return err
+	}
+	gid, err := strconv.Atoi(u.Gid)
+	if err != nil {
+		return err
+	}
+	runtimeConfig.HypervisorConfig.Uid = uint32(uid)
+	runtimeConfig.HypervisorConfig.Gid = uint32(gid)
+
+	userTmpDir := path.Join("/run/user/", fmt.Sprint(uid))
+	dir, err := os.Stat(userTmpDir)
+	if os.IsNotExist(err) {
+		if err = os.Mkdir(userTmpDir, vc.DirMode); err != nil {
+			return err
+		}
+		if err = syscall.Chown(userTmpDir, uid, gid); err != nil {
+			return err
+		}
+	}
+	if dir != nil && !dir.IsDir() {
+		return fmt.Errorf("%s is expected to be a directory", userTmpDir)
+	}
+
+	if err := os.Setenv("XDG_RUNTIME_DIR", userTmpDir); err != nil {
+		return err
+	}
+
+	info, _ := os.Stat("/dev/kvm")
+	var Gid uint32
+	if stat, ok := info.Sys().(*syscall.Stat_t); ok {
+		Gid = stat.Gid
+	} else {
+		return fmt.Errorf("failed to get the gid of /dev/kvm")
+	}
+	// Add the kvm group to the hypervisor supplemental group so that the hypervisor process can access /dev/kvm
+	runtimeConfig.HypervisorConfig.Groups = append(runtimeConfig.HypervisorConfig.Groups, Gid)
+
+	return nil
```

Comment:
this will crash on systems where kvm modules is not loaded, let's check the error returned by `os.Stat` and simplify the code a little bit more, wdyt?

```go
info, err := os.Stat("/dev/kvm")
if err != nil {
	return err
}

if stat, ok := info.Sys().(*syscall.Stat_t); ok {
	// Add the kvm group to the hypervisor supplemental group so that the hypervisor process can access /dev/kvm
	runtimeConfig.HypervisorConfig.Groups = append(runtimeConfig.HypervisorConfig.Groups, stat.Gid)
	return nil
}
return fmt.Errorf("failed to get the gid of /dev/kvm")
```